fadtk/model_loader.py

```diff
@@ -78,17 +85,110 @@ def load_model(self):
+        super().__init__(f"panns-{variant}", 2048, 
+                         sr=16000 if variant == 'cnn14-16k' else 32000, audio_len=audio_len)
+        self.variant = variant
+
+    def load_model(self):
+        current_file_dir = os.path.dirname(os.path.realpath(__file__))
+        ckpt_dir = os.path.join(current_file_dir, "panns/ckpt")
+        os.makedirs(ckpt_dir, exist_ok=True)
+        # check if each pth file is in ckpt_dir
+        if not os.path.isfile(os.path.join(ckpt_dir, "Cnn14_mAP=0.431.pth")):
+            print("Download pretrained checkpoints of Panns-Cnn14-32k.")
+            os.system(
+                f"wget -P {ckpt_dir} %s"
+                % ("https://zenodo.org/record/3576403/files/Cnn14_mAP%3D0.431.pth")
+            )
+        if not os.path.isfile(os.path.join(ckpt_dir, "Cnn14_16k_mAP=0.438.pth")):
+            print("Download pretrained checkpoints of Panns-Cnn14-16k.")
+            os.system(
+                f"wget -P {ckpt_dir} %s"
+                % ("https://zenodo.org/record/3987831/files/Cnn14_16k_mAP%3D0.438.pth")
+            )
+        if not os.path.isfile(os.path.join(ckpt_dir, "Wavegram_Logmel_Cnn14_mAP=0.439.pth")):
+            print("Download pretrained checkpoints of Panns-Wavegram-Logmel.")
+            os.system(
+                f"wget -P {ckpt_dir} %s"
+                % ("https://zenodo.org/records/3987831/files/Wavegram_Logmel_Cnn14_mAP%3D0.439.pth")
+            )
+        features_list = ["2048", "logits"]
+        current_file_dir = os.path.dirname(os.path.realpath(__file__))
+
+        if self.variant == 'cnn14-16k':
+            self.model = panns.Cnn14(
+                features_list=features_list,
+                sample_rate=16000,
+                window_size=512,
+                hop_size=160,
+                mel_bins=64,
+                fmin=50,
+                fmax=8000,
+                classes_num=527,
+            )
+            state_dict = torch.load(f"{current_file_dir}/panns/ckpt/Cnn14_16k_mAP=0.438.pth")
+            self.model.load_state_dict(state_dict["model"])
+
+        elif self.variant == 'cnn14-32k':
+            self.model = panns.Cnn14(
+                features_list=features_list,
+                sample_rate=32000,
+                window_size=1024,
+                hop_size=320,
+                mel_bins=64,
+                fmin=50,
+                fmax=14000,
+                classes_num=527,
+            )
+            state_dict = torch.load(f"{current_file_dir}/panns/ckpt/Cnn14_mAP=0.431.pth")
+            self.model.load_state_dict(state_dict["model"])
+
+        elif self.variant == 'wavegram-logmel':
+            self.model = panns.Wavegram_Logmel_Cnn14(
+                sample_rate=32000,
+                window_size=1024,
+                hop_size=320,
+                mel_bins=64,
+                fmin=50,
+                fmax=14000,
+                classes_num=527,
+            )
```

Comment:
add
```
else:
    raise ValueError(f"Unexpected variant: {self.variant}")
```

Reply:
fyi, because this is stronger and more helpful to the users than adding a type hint.

Reply:
this comment should go to line 169.